### DISCUSSION TITLE
Add a "maintenance mode" to Specify

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5995] [Feature Requests]
**Is your feature request related to a problem? Please describe.**
This request is similar to https://github.com/specify/specify7/issues/3619, but prevents other users from working in the database at all.

> Sometimes I need to work directly on the database, or I want to do a massive operation with the workbench, and I’d feel more peace of mind if I were to know that no one else is working on the system at the same time (though I know WB has been designed to address concurrent access, and I could stop the web server when working directly on the DB).
> 
> Institutional scoping might add complexity though, putting a whole institution on maintenance, or just a division or a discipline… :exploding_head:
> 
> But let’s say a maintenance mode at institution level, that would only accept logging from institution admin and show a “maintenance mode” screen to any other user.

**Describe the solution you'd like**
> ...including a “Maintenance mode” :hammer_and_wrench: entry in the [User Tools](https://discourse.specifysoftware.org/t/user-tools-menu/536) menu ?

**Reported By**
Philippe Verley at IRD via the [Speciforum](https://discourse.specifysoftware.org/t/maintenance-mode-in-specify-7/2092)

**Comment by specifysoftware** [comment]
This issue has been mentioned on **Specify Community Forum**. There might be relevant details there:

https://discourse.specifysoftware.org/t/maintenance-mode-in-specify-7/2092/2

**Comment by grantfitzsimmons** [comment]
From [Mark Pitblado](https://discourse.specifysoftware.org/t/maintenance-mode-in-specify-7/2092/3):
> To add to this, I am not sure how common it is with the various monitoring providers, but ours has the ability to add in an announcement banner into applications that syncs up to maintenance notifications on the status page.
> 
> Obviously not super critical, but a nice to have if easy to work into the implementation.
> 
> Script format for announcement integration
> 
> ```
> <script src="https://uptime.betterstack.com/widgets/announcement.js" data-id="<redacted>" async="async" type="text/javascript"></script>
> ```
